icecreamshop/src/main/java/com/summerschool/icecreamshop/controller/RateController.java

```diff
@@ -10,19 +10,43 @@
 import com.summerschool.icecreamshop.dto.RateDTO;
 import org.springframework.beans.factory.annotation.Autowired;
 import static com.summerschool.icecreamshop.exception.Constants.RATE_NOT_FOUND;
+import com.summerschool.icecreamshop.dto.RateDTO;
+import com.summerschool.icecreamshop.service.RateService;
+import com.summerschool.icecreamshop.model.Rate;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
 
+import javax.validation.Valid;
 import javax.validation.Valid;
 
 @RestController
 @RequestMapping("/ratings")
 public class RateController {
 
-    @Autowired
     private RateService rateService;
 
-    @Autowired
     private ModelMapper modelMapper;
 
+
+    @Autowired
+    public RateController(RateService rateService, ModelMapper modelMapper) {
+        this.rateService = rateService;
+        this.modelMapper = modelMapper;
+    }
+
+    @PostMapping
+    public ResponseEntity<RateDTO> add(@Valid @RequestBody RateDTO rateDTO) {
+        Rate rate = modelMapper.map(rateDTO, Rate.class);
+
+        Rate savedRate = rateService.add(rate);
```

Comment:
aici ar trebui sa verifici daca savedRate e null si daca e null sa returnezi un 422 Unprocessable Entity sau ceva asemanator in loc de CREATED.

Reply:
In legatura cu RateControler#add, l-am verificat in postman si functioneaza.
![badrequest](https://user-images.githubusercontent.com/72852566/131804883-ac81c895-015e-4baf-91d1-4654065c65a7.PNG)
](url)

Reply:
pentru ca ai trimis un format gresit.  daca introduci formatul jsonul corect va returna ACCEPTED indiferent daca obiectul se creeaza in baza de date cu success sau nu.